contrib/babelfishpg_tsql/sql/sys_cast.sql

```diff
@@ -168,6 +168,22 @@ RETURNS ANYELEMENT
 AS $BODY$ BEGIN
     EXECUTE pg_catalog.format('SELECT CAST(%L AS %s)', arg, format_type(pg_typeof(output), typmod)) INTO output;
     EXCEPTION
+        WHEN cannot_coerce THEN
+            RAISE USING MESSAGE := pg_catalog.format('cannot cast type text to %s.',
+                                      pg_typeof(output));
+        WHEN OTHERS THEN
+            -- Do nothing. Output carries NULL.
+END; $BODY$
+LANGUAGE plpgsql;
+
+CREATE OR REPLACE FUNCTION sys.babelfish_try_cast_to_any(IN arg ANYCOMPATIBLE, INOUT output ANYELEMENT, IN typmod INT)
```

Comment:
I think initially when I only had the function for ANYCOMPATIBLE if you ran a query like "TRY_CAST('1990-01-01' as date)" it would not enter the try_cast_to_any function because it didnt have an explicit type.